postreise/analyze/generation/summarize.py

```diff
@@ -129,3 +133,42 @@ def summarize_hist_gen(hist_gen_raw: pd.DataFrame, all_resources: list) -> pd.Da
     result.rename(columns=type2label, inplace=True)
 
     return result
+
+
+def get_generation_time_series_by_resources(scenario, area, resources, area_type=None):
+    """Get time series generation for each resource in certain area of a scenario
+
+    :param powersimdata.scenario.scenario.Scenario scenario: scenario instance
+    :param str area: one of *loadzone*, *state*, *state abbreviation*,
+        *interconnect*, *'all'*
+    :param str/list resources: one or a list of resources
+    :param str area_type: one of *'loadzone'*, *'state'*, *'state_abbr'*,
+        *'interconnect'*
+    :return: (*pandas.DataFrame*) -- times series of generation for each resource,
+        index: time stamps, columns: resources
+    """
+    _check_scenario_is_in_analyze_state(scenario)
+    plant_id = get_plant_id_for_resources_in_area(
+        scenario, area, resources, area_type=area_type
+    )
+    pg = scenario.state.get_pg()[plant_id]
+    grid = scenario.state.get_grid()
+
+    return pg.groupby(grid.plant.loc[plant_id, "type"].values, axis=1).sum()
+
+
+def get_storage_time_series(scenario, area, area_type=None):
```

Comment:
Should we call this something like `get_storage_pg_time_series`, or have another parameter that specifies whether we want `storage_pg` or `storage_e`? This could easily be in a different PR though, we don't need to hold this one up.

Reply:
Done in #215.